src/main/java/com/taboola/backstage/Backstage.java

```diff
@@ -176,15 +186,17 @@ public Backstage build() {
             organizeState();
             CommunicationConfig config = new CommunicationConfig(baseUrl, authBaseUrl, connectionTimeoutMillis, readTimeoutMillis, writeTimeoutMillis, userAgent, debug);
             CommunicationFactory communicator = new CommunicationFactory(config);
+            BackstageEndpointsFactory endpointsFactory = new BackstageEndpointsFactoryImpl(communicator);
```

Comment:
I wasn't able to understand how one will be able to give different implementation to an endpoint.
in other words, how we will use it in our GW to give different impl ?